types/events.go

```diff
@@ -137,6 +165,37 @@ func ParseTypedEvent(event abci.Event) (proto.Message, error) {
 	return protoMsg, nil
 }
 
+// ResultEventToABCIEvent takes the ctypes.ResultEvent and casts it to abci.TxResult, extracting the []abci.Event
+func ResultEventToABCIEvent(rev ctypes.ResultEvent) ([]abci.Event, error) {
+	switch rev.Query {
+	case txEvents:
+		var txResult abci.TxResult
+		txResBytes, err := json.Marshal(rev.Data)
+		if err != nil {
+			return nil, err
+		}
+		if err := json.Unmarshal(txResBytes, &txResult); err != nil {
```

Comment:
Why are we doing this marshalling and unmarshalling?

Can't we just cast `rev.Data` to `tmtypes.EventDataTx`?

And then we can get the `abci.TxResult` inside?

https://github.com/tendermint/tendermint/blob/master/rpc/core/types/responses.go#L233
https://github.com/tendermint/tendermint/blob/e13b4386ff1a46a5e8879e11bd653b62d4ca0992/types/events.go#L86